lib/rules/menu-item-needs-labelling-v9.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+"use strict";
+
+const { hasNonEmptyProp } = require("../util/hasNonEmptyProp");
+var elementType = require("jsx-ast-utils").elementType;
+const { hasAssociatedLabelViaAriaLabelledBy, isInsideLabelTag, hasAssociatedLabelViaHtmlFor } = require("../util/labelUtils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+    meta: {
+        // possible error messages for the rule
+        messages: {
+            noUnlabelledMenuItem: "Accessibility: MenuItem must have an accessible label"
+        },
+        // "problem" means the rule is identifying code that either will cause an error or may cause a confusing behavior: https://eslint.org/docs/latest/developer-guide/working-with-rules
+        type: "problem",
+        docs: {
+            description: "Accessibility: MenuItem without label must have an accessible and visual label: aria-labelledby",
+            recommended: true,
+            url: "https://www.w3.org/TR/html-aria/" // URL to the documentation page for this rule
+        },
+        fixable: null, // Or `code` or `whitespace`
+        schema: [] // Add a schema if the rule has options
+    },
+
+    create(context) {
+        return {
+            // visitor functions for different types of nodes
+            JSXOpeningElement(node) {
+                // if it is not a MenuItem, return
+                if (elementType(node) !== "MenuItem") {
+                    return;
+                }
+
+                // if the MenuItem has a label, if the MenuItem has an associated label, return
+                if (
+                    hasNonEmptyProp(node.attributes, "aria-label") || //aria-label, not recommended but will work for screen reader users
+                    isInsideLabelTag(context) || // wrapped in label
+                    hasAssociatedLabelViaHtmlFor(node, context) || // label with htmlFor
```

Comment:
I need to double check but I think htmlFor is just for inputs but I need to verify

Reply:
removed , i verified it , its not required